miio/cooker_multi.py

```diff
@@ -0,0 +1,419 @@
+        if not len(data) % 2:
+            self.data = [int(data[i : i + 2], 16) for i in range(0, len(data), 2)]
+        else:
+            self.data = []
+
+    @property
+    def temperatures(self) -> List[int]:
+        return self.data
+
+    @property
+    def raw(self) -> str:
+        return "".join([f"{value:02x}" for value in self.data])
+
+    def __str__(self) -> str:
+        return str(self.data)
+
+
+class MultiCookerProfile:
+    """This class can be used to modify and validate an existing cooking profile."""
+
+    def __init__(self, profile_hex: str, duration: int, schedule: int, akw: bool):
+        if len(profile_hex) < 5:
+            raise CookerException("Invalid profile")
+        else:
+            self.checksum = bytearray.fromhex(profile_hex)[-2:]
+            self.profile_bytes = bytearray.fromhex(profile_hex)[:-2]
+
+            if not self.is_valid():
+                raise CookerException("Profile checksum error")
+
+            self.set_schedule_enabled(False)
+
+            if duration is not None:
+                self.set_duration(duration)
+            if schedule is not None and schedule > 0 and schedule <= 1440:
+                self.set_schedule_enabled(True)
+                self.set_schedule_duration(schedule)
+            if akw is not None:
+                self.set_akw_enabled(akw)
+
+    def is_set_duration_allowed(self):
+        return (
+            self.profile_bytes[10] != self.profile_bytes[12]
+            or self.profile_bytes[11] != self.profile_bytes[13]
+        )
+
+    def get_duration(self):
+        """Get the duration in minutes."""
+        return (self.profile_bytes[8] * 60) + self.profile_bytes[9]
+
+    def set_duration(self, minutes):
+        """Set the duration in minutes if the profile allows it."""
+        if not self.is_set_duration_allowed():
+            return
+
+        max_minutes = (self.profile_bytes[10] * 60) + self.profile_bytes[11]
+        min_minutes = (self.profile_bytes[12] * 60) + self.profile_bytes[13]
+
+        if minutes < min_minutes or minutes > max_minutes:
+            return
+
+        self.profile_bytes[8] = math.floor(minutes / 60)
+        self.profile_bytes[9] = minutes % 60
+
+        self.update_checksum()
+
+    def is_schedule_enabled(self):
+        return (self.profile_bytes[14] & 0x80) == 0x80
+
+    def set_schedule_enabled(self, enabled):
+        if enabled:
+            self.profile_bytes[14] |= 0x80
+        else:
+            self.profile_bytes[14] &= 0x7F
+
+        self.update_checksum()
+
+    def set_schedule_duration(self, duration):
+        """Set the schedule time (delay before cooking) in minutes."""
+        schedule_flag = self.profile_bytes[14] & 0x80
+        self.profile_bytes[14] = math.floor(duration / 60) & 0xFF
+        self.profile_bytes[14] |= schedule_flag
+        self.profile_bytes[15] = (duration % 60 | self.profile_bytes[15] & 0x80) & 0xFF
+
+        self.update_checksum()
+
+    def is_akw_enabled(self):
```

Comment:
Please add docstrings to all public API. Also, is there a better name for `akw` or is this somewhat common knowledge among users of these devices?

Reply:
`auto_keep_warm` is longer but more descriptive, i had to look up what `akw` was

Reply:
changed